skimage/transform/_warps.py

```diff
@@ -106,17 +107,20 @@ def resize(image, output_shape, order=None, mode='reflect', cval=0, clip=True,
                          "dimensions")
 
     if anti_aliasing is None:
-        anti_aliasing = not image.dtype == bool
+        anti_aliasing = (not input_type == bool and
+                         any(x < y for x,y in zip(output_shape, input_shape)))
 
-    if image.dtype == bool and anti_aliasing:
+    if input_type == bool and anti_aliasing:
         warn("Input image dtype is bool. Gaussian convolution is not defined "
              "with bool data type. Please set anti_aliasing to False or "
              "explicitely cast input image to another data type. Starting "
              "from version 0.19 a ValueError will be raised instead of this "
              "warning.", FutureWarning, stacklevel=2)
 
-    factors = (np.asarray(input_shape, dtype=float) /
-               np.asarray(output_shape, dtype=float))
+    factors = np.divide(input_shape, output_shape)
+    image = convert_to_float(image, preserve_range)
+    # create copy so input values range stays accessible through image for clip
```

Comment:
```suggestion
    # create copy so input value range stays accessible through image for clip
```
No big deal, but it would be more intuitive for `img_in` to be the copy used for retaining the range of values of the input image (both because `img_in` means, I guess, 'input image' and because `img_in` lies at the LHS of the equal sign). So, I would expect `img_in` to keep aside and `image` to undergo all subsequent operations.